src/sentry/search/django/backend.py

```diff
@@ -250,6 +257,80 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        # TODO: This could also do something janky like switching based on the current
+        #       tagstore backend, or a global option, or a number of other things. Going
+        #       by the project seemed the best for now.
+        use_snuba = bool(project.get_option('sentry:snuba', False))
+        if use_snuba and (tags or
+                          environment is not None or
+                          any(key in parameters for key in ('date_from', 'date_to'))):
+
+            # if we're in this branch then Snuba can help us filter down the eligible group ids
+            filters = {
+                'project_id': [project.id],
+            }
+
+            if environment is not None:
+                filters['environment'] = [environment.id]
```

Comment:
I remember how this works now (rewritten through `snuba.query`, etc.), but this caught me off guard for a second because I thought we were storing environments by name in the database.

Reply:
Ya I actually made that mistake myself. It's a shame we go id -> query -> name but we discussed in other PR and we can improve later.